src/NuGet.Core/NuGet.Common/AuthTypeFilteredCredentials.cs

```diff
@@ -3,45 +3,44 @@
 
 using System;
 using System.Collections.Generic;
-using System.Net;
 using System.Linq;
+using System.Net;
 
-
-namespace NuGet.Credentials
+namespace NuGet.Common
 {
     /// <summary>
-    /// Wraps another ICredentials object by returning null if the authType is not in a specified allow list
+    /// Wraps a NetworkCredentials object by returning null if the authType is not in a specified allow list
     /// </summary>
     public class AuthTypeFilteredCredentials : ICredentials
     {
-        private readonly IList<string> _authTypes;
-        private readonly ICredentials _innerCredential;
+        public IReadOnlyList<string> AuthTypes { get; }
+        public NetworkCredential InnerCredential { get; }
 
         /// <summary>
         /// Initializes a new AuthTypeFilteredCredentials
         /// </summary>
         /// <param name="innerCredential">Credential to delegate to</param>
-        /// <param name="authTypes">List of authTypes to respond to. May not be null or empty.</param>
-        public AuthTypeFilteredCredentials(ICredentials innerCredential, IList<string> authTypes)
+        /// <param name="authTypes">List of authTypes to respond to. If empty, responds to all authTypes.</param>
+        public AuthTypeFilteredCredentials(NetworkCredential innerCredential, IEnumerable<string> authTypes)
         {
             if (innerCredential == null)
             {
                 throw new ArgumentNullException(nameof(innerCredential));
             }
-            
-            if (authTypes == null || !authTypes.Any())
+
+            if (authTypes == null)
             {
-                throw new ArgumentException(Resources.Error_Argument_May_Not_Be_Null_Or_Empty, nameof(authTypes));
+                throw new ArgumentNullException(nameof(authTypes));
             }
 
-            _innerCredential = innerCredential;
-            _authTypes = authTypes;
+            InnerCredential = innerCredential;
+            AuthTypes = new List<string>(authTypes);
         }
 
         public NetworkCredential GetCredential(Uri uri, string authType)
         {
-            return _authTypes.Any(x => StringComparer.OrdinalIgnoreCase.Equals(x, authType))
-                ? _innerCredential.GetCredential(uri, authType)
+            return authType == null || !AuthTypes.Any() || AuthTypes.Any(x => StringComparer.OrdinalIgnoreCase.Equals(x, authType))
```

Comment:
Why change this?

Reply:
NuGet has elsewhere assumed that you can call GetCredential with an authType of null, and the null comparison allows that to pass through. The !AuthTypes.Any() check moves the responsibility of determining that "an empty filter means allow all" from the caller where it doesn't belong, to this class, where it does.

Reply:
ok, makes sense.